src/main/webapp/app/components/sidebar/SideBar.tsx

```diff
@@ -0,0 +1,65 @@
+import { IconProp } from '@fortawesome/fontawesome-svg-core';
+import { faArrowDown, faBars, faCartArrowDown, faChevronDown, faDna, faLongArrowAltDown } from '@fortawesome/free-solid-svg-icons';
+import { faSearch } from '@fortawesome/free-solid-svg-icons/faSearch';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { PAGE_ROUTE } from 'app/config/constants';
+import { IRootStore } from 'app/stores/createStore';
+import { componentInject } from 'app/shared/util/typed-inject';
+import { observer } from 'mobx-react';
+import React from 'react';
+import { NavLink } from 'react-router-dom';
+import { Nav } from 'reactstrap';
+import './sidebar.scss';
+
+type SideBarMenuItemProps = {
+  icon: IconProp;
+  name: string;
+  linkTo?: string;
+  onClick?: () => void;
+};
+
+const SideBarMenuItem: React.FunctionComponent<SideBarMenuItemProps> = props => {
+  const menuItem = (
+    <div className="menu-item" onClick={props.onClick}>
+      <FontAwesomeIcon icon={props.icon} size="lg" className="mr-2" />
+      <span className="menu-text">{props.name}</span>
+    </div>
+  );
+  if (props.linkTo) {
+    return (
+      <NavLink to={props.linkTo} className="nav-item" exact>
+        {menuItem}
+      </NavLink>
+    );
+  }
+  return menuItem;
+};
+
+class SideBar extends React.Component<StoreProps> {
```

Comment:
We can start with the simple implementation, but later when we start to implement the submenu, could consider https://github.com/azouaoui-med/react-pro-sidebar or other options.

Reply:
Another option is https://www.devwares.com/docs/contrast/react/navigation/sidebar/. I will try both out.